documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc

```diff
@@ -177,9 +277,41 @@ include::sds-developer-guide:ROOT:partial$sds-aspect-model-serializer-artifact.a
 [[validating-aspect-models]]
 == Validating Aspect Models
 
+Aspect Models are validated using the `AspectModelValidator`. Validation returns a list of
+`Violation`&#8203;s. A violation has a human-readable message and a unique error code and provides access
+to the `EvaluationContext` which contains references to the model element that caused the
+violation and the SHACL shape that triggered the violation.
+
+Each possible type of violation is a subtype of the `Violation` interface and provides additional
+context information specific to this type, for example, the `MinLengthViolation` provides `int min`
+(the allowed length) and `int actual` (the length that was encountered in the model).
+
 Consider the following example:
 
-include::sds-developer-guide:ROOT:partial$sample-validating-aspect-model.adoc[]
+++++
+<details>
+<summary>Show used imports</summary>    
+++++
+[source,java,indent=0,subs="+macros,+quotes"]
+----
+include::example$ValidateAspectModel.java[tags=imports]
+----
+++++
+</details>
+++++
+
+[source,java,indent=0,subs="+macros,+quotes"]
+----
+include::example$ValidateAspectModel.java[tags=validate]
+----
+
+<1> For an immediate human-readable result of the validation result, use the `ViolationFormatter` or
```

Comment:
... result of the validation result ... sounds a little strange, is there a better formulation?